### DISCUSSION TITLE
When should the "Source change" box display?

### DISCUSSION THREAD

**Discussion by brockfanning** [WeblateOrg/weblate#7319] [Q&A] [answered]
When using a monolingual format, if the source for a string changes, then a "Source change" box appears for the translators, showing a diff of what changed. This is amazing functionality and is very useful to translators. However I've heard reports from translators that the "Source change" box is sometimes missing, at times when they are expecting to see it. I would like to dig into these reports a bit more and do some testing, but first I was hoping to get some clarity from the community:

How does Weblate know when to show -- and when to *stop* showing -- the "Source change" box?

My guess would be that it starts showing as soon as the source file changes (pretty straightforward) but I am not sure how it would know when to stop showing it.

For example, take this source file with plurality mistakes, "en/fruit.yml":

```
apple: Apples
banana: Bananas
```

And this French translation of "fr/fruit.yml" which includes the plurality mistakes:

```
apple: Pommes
banana: Bananes
```

If "en/fruit.yml" changes to singular instead of plural...

```
apple: Apple
banana: Banana
```

...then the "Source change" box should start to appear on the French translation pages for both strings.

Then, a translator updates the French translation of the "apple" string to "Pomme" in Weblate.

* Is that the point at which the "Source change" box stops appearing on the French translation page for the "apple" string?
* If not, what are the criteria for when Weblate stops showing that "Source change" box?
* When the "Source change" box stops appearing on the French translation page for the "apple" string, will it continue to show for the French translation page for the "banana" string?

Hope that all makes sense. Thank you for any help!

**Comment by nijel** [comment, marked as answer]
For monolingual formats, it behaves as you described. For bilingual formats, this information has to be present in the file.
